# Chi-Square Test for Association between Device Type and Customer Satisfaction

### Hypotheses:
**Null Hypothesis (H₀)**: There is no significant association between the type of device (Smart Thermostats vs. Smart Lights) and customer satisfaction.
**Alternative Hypothesis (H₁)**: There is a significant association between the type of device and customer satisfaction.

### Step 1: Observed Frequencies
The observed frequencies from the contingency table are:

In [1]:
import numpy as np
import scipy.stats as stats

# Observed frequencies from the contingency table
observed = np.array([[50, 70],
                     [80, 100],
                     [60, 90],
                     [30, 50],
                     [20, 50]])
observed

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

### Step 2: Chi-Square Statistic
We will compute the Chi-Square statistic, degrees of freedom, and expected frequencies:

In [2]:
# Calculate the Chi-Square statistic and p-value using scipy
chi2_stat, p_val, dof, expected = stats.chi2_contingency(observed)

# Output the Chi-Square statistic, degrees of freedom, and expected frequencies
chi2_stat, p_val, dof, expected

(5.638227513227513,
 0.22784371130697179,
 4,
 array([[ 48.,  72.],
        [ 72., 108.],
        [ 60.,  90.],
        [ 32.,  48.],
        [ 28.,  42.]]))

### Step 3: Critical Value
Using a significance level of 0.05 and the degrees of freedom (df), we determine the critical value:

In [3]:
# Determine the critical value at significance level 0.05
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, dof)
critical_value

9.487729036781154

### Step 4: Conclusion
If the Chi-Square statistic is greater than the critical value, we reject the null hypothesis. Otherwise, we fail to reject it.

**Chi-Square Statistic**: 5.64
**Critical Value**: 9.49
**p-value**: 0.228

Since the Chi-Square statistic is less than the critical value and the p-value is greater than 0.05, we fail to reject the null hypothesis.

Thus, there is no significant association between the type of device and customer satisfaction.